tests/ui/eta.stderr

```diff
@@ -31,132 +31,138 @@ LL |     let e = Some(1u8).map(|a| generic(a));
    |                           ^^^^^^^^^^^^^^ help: replace the closure with the function itself: `generic`
 
 error: redundant closure
-  --> $DIR/eta.rs:87:51
+  --> $DIR/eta.rs:54:10
+   |
+LL |     test(|start, end| start..=end);
+   |          ^^^^^^^^^^^^^^^^^^^^^^^^ help: replace the closure with the function itself: `core::ops::RangeInclusive::new`
+
```

Comment:
IMO we shouldn't lint this case, it feels a bit too odd to suggest even though it would be equivalent